document/src/vespa/document/update/tensormodifyupdate.cpp

```diff
@@ -0,0 +1,136 @@
+// Copyright 2019 Yahoo Holdings. Licensed under the terms of the Apache 2.0 license. See LICENSE in the project root.
+
+#include "tensormodifyupdate.h"
+#include <vespa/document/base/field.h>
+#include <vespa/document/base/exceptions.h>
+#include <vespa/document/fieldvalue/tensorfieldvalue.h>
+#include <vespa/document/util/serializableexceptions.h>
+#include <vespa/eval/tensor/serialization/typed_binary_format.h>
+#include <vespa/eval/tensor/tensor.h>
+#include <vespa/vespalib/objects/nbostream.h>
+#include <vespa/vespalib/stllike/asciistream.h>
+#include <vespa/vespalib/util/stringfmt.h>
+#include <vespa/vespalib/util/xmlstream.h>
+#include <ostream>
+
+using vespalib::IllegalArgumentException;
+using vespalib::IllegalStateException;
+using vespalib::tensor::TypedBinaryFormat;
+using vespalib::tensor::Tensor;
+using vespalib::make_string;
+
+namespace document {
+
+IMPLEMENT_IDENTIFIABLE(TensorModifyUpdate, ValueUpdate);
+
+TensorModifyUpdate::TensorModifyUpdate()
+    : _operator(Operator::MAX_NUM_OPERATORS),
+      _operand()
+{
+}
+
+TensorModifyUpdate::TensorModifyUpdate(const TensorModifyUpdate &rhs)
+    : _operator(rhs._operator),
+      _operand(rhs._operand->clone())
+{
+}
+
+TensorModifyUpdate::TensorModifyUpdate(Operator op, std::unique_ptr<vespalib::tensor::Tensor> &&operand)
+    : _operator(op),
+      _operand(std::move(operand))
+{
+}
+
+TensorModifyUpdate::~TensorModifyUpdate() = default;
+
+TensorModifyUpdate &
+TensorModifyUpdate::operator=(const TensorModifyUpdate &rhs)
+{
+    _operator = rhs._operator;
+    _operand = rhs._operand->clone();
+    return *this;
+}
+
+TensorModifyUpdate &
+TensorModifyUpdate::operator=(TensorModifyUpdate &&rhs)
+{
+    _operator = rhs._operator;
+    _operand = std::move(rhs._operand);
+    return *this;
+}
+
+bool
+TensorModifyUpdate::operator==(const ValueUpdate &other) const
+{
+    if (other.getClass().id() != TensorModifyUpdate::classId) return false;
```

Comment:
I prefer usage of curly braces for if-statements and not putting them on a single line. The same applies for lines 67 and 68.